apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/Vesting.tsx

```diff
@@ -1,7 +1,59 @@
-import React, { FC } from 'react';
+import React, { FC, useCallback, useMemo, useState } from 'react';
 
-export const Vesting: FC = () => (
-  <>
-    <div>TBD - Vesting rewards table</div>
-  </>
-);
+import { t } from 'i18next';
+
+import { Pagination, Table } from '@sovryn/ui';
+
+import { DEFAULT_PAGE_SIZE } from '../../../../../constants/general';
+import { useAccount } from '../../../../../hooks/useAccount';
+import { translations } from '../../../../../locales/i18n';
+import { columnsConfig } from './Vestings.constants';
+import { generateRowTitle } from './Vestings.utils';
+import { useGetVestingContracts } from './hooks/useGetVestingContracts';
+
+const pageSize = DEFAULT_PAGE_SIZE;
+
+export const Vesting: FC = () => {
+  const [page, setPage] = useState(0);
+  const { data: vestings, loading } = useGetVestingContracts(page, pageSize);
+  const { account } = useAccount();
+
+  const onPageChange = useCallback(
+    (value: number) => {
+      if (!vestings || (vestings?.length < pageSize && value > page)) {
+        return;
+      }
+      setPage(value);
+    },
+    [page, vestings],
+  );
+
+  const isNextButtonDisabled = useMemo(
+    () => !loading && (!vestings || vestings.length < pageSize),
+    [loading, vestings],
+  );
+
+  return (
+    <div className="lg:bg-gray-80 lg:py-4 lg:px-4 rounded w-full">
+      <Table
+        columns={columnsConfig}
+        rows={vestings}
+        noData={
+          account
+            ? t(translations.rewardPage.vesting.table.noRewards)
+            : t(translations.rewardPage.vesting.table.notConnected)
+        }
+        dataAttribute="funding-history-table"
+        rowTitle={generateRowTitle}
+      />
+      <Pagination
```

Comment:
we should hide pagination if there are no vestings available or the user is not connected

Reply:
We don't do it in the history page either so I wanted to stay consistent.

Reply:
yep, makes sense